src/walk.rs

```diff
@@ -48,11 +48,12 @@ pub enum FileType {
 /// If the `--exec` argument was supplied, this will create a thread pool for executing
 /// jobs in parallel from a given command line and the discovered paths. Otherwise, each
 /// path will simply be written to standard output.
-pub fn scan(root: &Path, pattern: Arc<Regex>, config: Arc<FdOptions>) {
+pub fn scan(path_vec: &mut Vec<PathBuf>, pattern: Arc<Regex>, config: Arc<FdOptions>) {
```

Comment:
Using a mutable reference here seems strange to me. Could `scan` not be implemented without using `pop`?